### ISSUE TITLE
Compilation process removed added <a> tag attribute

### ISSUE THREAD

**Issue by davehoran** [foundation/foundation-emails#776]
My employer is using Localytics (https://www.localytics.com/) as their platform of choice for sending emails to customers. As most of the users are on mobile devices, the emails contain deep links into the app that is (presumably) installed on their device. 

In order to get this deep linked to work, and additional attribute is needed to get <a> tags to link into the app properly. Here is an example link:

`<a href="https://v0eq.app.link/DuNztpZLpD" universal="true">Shop</a>`

When the Inky code is compiled, the attribute **universal="true"** is stripped out of the anchor tag. I needed, and expected, that this attribute would have been passed through and show up in the rendered HTML/CSS code. 

Based on what I've been able to trace through, this seems to be happening before the inlining process. Is there a known list of attributes that the system will not strip out? If so, how can I alter this?

**How can we reproduce this bug?**

Create an anchor tag with the additional attribute universal="true" . Let the gulp process run to compile the code, then look for the can attribute in the rendered HTML. If it not there, you've successfully reproduced the bug. 

Write out the HTML (or Inky code) that causes the issue.
`<a href="https://v0eq.app.link/DuNztpZLpD" universal="true">Shop</a>`

**What did you expect to happen?**
I expected the attribute **universal="true"** to be passed through from the source Inky code to the rendered HTML/CSS code. 

**What happened instead?**
The attribute was stripped out.

**What email clients does this happen in?**
This is not related to an email client. 

**Edit**:
I have discovered that if I add the `universal="true"` in an `<a>` tag, it stays, but NOT when added as an attribute to a button. No sure what's up.

**Comment by davehoran** [comment]
Does anyone have an idea where to look? Where in the system is the HTML being parsed and "fixed"? This fixing is breaking the carefully-added tag attributes. I'll dive in, but I need to know where in the ocean to start. Thanks.

**Comment by davehoran** [comment]
The only way I was able to get around this was kinda drastic. I modified the /node_modules/inky/lib/componentFactory.js file and modified the output for the anchor tag in the <button> interpolation:
`// If we have the href attribute we can create an anchor for the inner of the button;
      if (element.attr('href')) {
        inner = format('<a href="%s"%s universal="true">%s</a>', element.attr('href'), target, inner);
      }
`
Now any button rendering will contain the universal="true" attribute in the output HTML. Not an idea solution, but I couldn't figure out a simpler way with a helper function.
